lib/AST/Identifier.cpp

```diff
@@ -20,13 +20,21 @@
 #include "llvm/Support/ConvertUTF.h"
 using namespace swift;
 
+void *DeclBaseName::SubscriptIdentifierData =
```

Comment:
Is there a quick explanation of why the cast is illegal or somewhere I can look it up? Proves once again that I know less about C++ than I thought ;-)

I wouldn't really want to change away from using `Identifier` as the underlying storage type of `DeclBaseName`. Somehow it doesn't feel right to me and would imho lead to unnecessary conversions whenever `getIdentifier` is called on `DeclBaseName`. Or did you mean changing the `char *` in `Identifier` to `void *`.

I decided against using a constant string because of the following (I might be wrong): Say we set `SubscriptIdentifierData` to "specialSubscript". Then, if somebody later decided to use a constant string named "specialSubscript" somewhere completely different it may end up being stored at the same location in the binary. Hence if you create an `Identifier` with this constant name it would be a special name. (But as I'm writing this, I realise that the chances of that are pretty low 😉)

Also: why would the string need to be 4-byte aligned?